File: discord/src/main/java/io/github/shaksternano/borgar/discord/media/io/writer/MediaWriter.java
package io.github.shaksternano.borgar.discord.media.io.writer;

import io.github.shaksternano.borgar.discord.media.AudioFrame;
import io.github.shaksternano.borgar.discord.media.ImageFrame;

import java.io.Closeable;
import java.io.IOException;

public interface MediaWriter extends Closeable {

    void writeImageFrame(ImageFrame frame) throws IOException;

    void writeAudioFrame(AudioFrame frame) throws IOException;

    boolean isStatic();

    boolean supportsAudio();
}

File: core/src/main/kotlin/media/writer/MediaWriter.kt
package com.shakster.borgar.core.media.writer

import com.shakster.borgar.core.io.SuspendCloseable
import com.shakster.borgar.core.media.AudioFrame
import com.shakster.borgar.core.media.ImageFrame

interface MediaWriter : SuspendCloseable {

    val isStatic: Boolean
    val supportsAudio: Boolean

    suspend fun writeImageFrame(frame: ImageFrame)

    suspend fun writeAudioFrame(frame: AudioFrame)
}
